pkgs/development/yottadb/default.nix

```diff
@@ -0,0 +1,168 @@
+      --set-default ydb_dist $out/dist/utf8 \
+      --set-default ydb_chset UTF-8 \
+      --set-default ydb_icu_version $(icu-config --version) \
+      --set-default LANG en_US.UTF-8 \
+      --set-default LC_ALL en_US.UTF-8
+
+    cd $out/dist
+    ln -rfs yottadb mumps
+    for utl in dse ftok gtcm_gnp_server gtcm_pkdisp gtcm_play gtcm_server gtcm_shmclean gtmsecshr lke mupip semstat2 yottadb mumps ; do
+      ln -rfs $utl ../bin/
+      ln -rfs $utl ./utf8/
+    done
+    ln -rfs libyottadb.so ../lib/
+    for inc in $(find -maxdepth 1 -mindepth 1 -name '*.h') ; do
+      ln -rfs $inc ../include/
+      ln -rfs $inc ./utf8/
+    done
+
+    sed -e s,@@YDB_DIST@@,$out/dist,g \
+        -e s,@@YDB_EXEDIR@@,$out/bin,g \
+        -e s,@@YDB_INCDIR@@,$out/include,g \
+        -e s,@@YDB_LIBDIR@@,$out/lib,g \
+        ${yottadbPkgConfigTemplate} > $out/lib/pkgconfig/yottadb.pc
+
+    faketty () {
+      script -qefc "$(printf "%q " "$@")" /dev/null
+    }
+
+    # Embed source to OBJs, so that we can drop SRCs
+    export ydb_compile="-embed_source -noignore"
+
+    # Build M-mode dist
+    export ydb_dist=$out/dist ydb_chset=M
+    cd $ydb_dist
+    faketty ./yottadb -r %XCMD 'zhalt $s($zyre["${version}":0,1:127)&$s($zch="M":0,1:126)'
+    faketty ./yottadb -r %XCMD 'd SILENT^%RSEL("*","SRC") s x="" f  s x=$o(%ZR(x)) q:x=""  w "["_$zch_"] Building "_x_" ...",! zl $tr(x,"%","_")'
+
+    # Build UTF8-mode dist
+    for rtn in *.m *.gld *.dat ; do
+      ln -rfs $rtn utf8/
+    done
+    unset ydb_chset ydb_icu_version ydb_dist
+    cd $out/dist/utf8
+    faketty ./yottadb -r %XCMD 'zhalt $s($zyre["${version}":0,1:127)&$s($zch="UTF-8":0,1:126)'
+    faketty ./yottadb -r %XCMD 'd SILENT^%RSEL("*","SRC") s x="" f  s x=$o(%ZR(x)) q:x=""  w "["_$zch_"] Building "_x_" ...",! zl $tr(x,"%","_")'
+
+    # Wipe SRCs as Nix is read-only and sets both .o and .m to the same mtime, so compiler tries to compile all the time, but cannot write
+    rm -f $out/dist/*.m $out/dist/utf8/*.m
+
+    # Also generate a single libyottadbutil.so to consolidate all system routines
+    libYdbUtil=libyottadbutil
+    isarmYdb=$(file $ydb_dist/yottadb | grep -wc "ARM" || true)
+    ldflags="-shared"
+    if [ "$isarmYdb" -eq 1 ] ; then
+      ldcmd="cc"  # Linux on ARM builds shared ELFs only using 'cc', not 'ld'
+    else
+      ldcmd="ld.gold"
+    fi
+    cd $out/dist
+    $ldcmd $ldflags -o $libYdbUtil.so *.o
+    ln -rfs $libYdbUtil.so ../lib/$libYdbUtil-m.so
+    cd $out/dist/utf8
+    $ldcmd $ldflags -o $libYdbUtil.so *.o
+    ln -rfs $libYdbUtil.so ../../lib/$libYdbUtil-utf8.so
+
+    # Default is UTF-8
+    cd $out/lib
+    ln -rfs $libYdbUtil-utf8.so $libYdbUtil.so
+  '';
+
+  meta = with lib; {
+    description = "A proven Multi-Language NoSQL database engine";
+    homepage = "https://gitlab.com/YottaDB/DB/YDB";
+    license = licenses.agpl3;
+    maintainers = [ maintainers.ztmr ];
+    platforms = [ "aarch64-linux" "armv7l-linux" "armv6l-linux" "x86_64-linux" ];
```

Comment:
this can probably just be `platforms.linux`

Reply:
I am not so sure about this. YottaDB is a specific piece of software -- it has some parts written in assembler for its target platform and was once supported on various platforms like VAX, Alpha, SPARC, HP9000, and x86, x86_64 were added later, and ARM-flavours were added only recently. There is also RISC-V merge request that might be merged eventually...
So I think it makes sense to list these supported platforms explicitly...